pode/utils.py

```diff
@@ -1,100 +1,233 @@
+        if segment_to_start.length < segment_to_end.length:
+            return [start, end]
+        else:
+            result = vertices[start_index + 1:-1] + vertices[:end_index + 1]
+    head = [vertices[start_index]] if start in vertices else [start]
+    tail = [] if end in vertices else [end]
+    return head + result + tail
+
+
+def edges(contour: Contour) -> Iterator[Segment]:
+    vertices = [*contour.vertices, contour.vertices[0]]
+    yield from (Segment(vertices[index], vertices[index + 1])
+                for index in range(len(vertices) - 1))
+
+
+def shrink_collinear_vertices(contour: Contour) -> List[Point]:
+    vertices = list(contour.vertices)
+    index = -len(vertices) + 1
+    while index < 0:
+        while (max(2, -index) < len(vertices)
+               and (Contour([vertices[index + 2],
+                             vertices[index + 1],
+                             vertices[index]]).orientation
+                    is Orientation.COLLINEAR)):
+            del vertices[index + 1]
+        index += 1
+    while index < len(vertices):
+        while (max(2, index) < len(vertices)
+               and (Contour([vertices[index - 2],
+                             vertices[index - 1],
+                             vertices[index]]).orientation
+                    is Orientation.COLLINEAR)):
+            del vertices[index - 1]
+        index += 1
+    return vertices
+
+
+@singledispatch
+def to_fractions(geometry: GeometryType) -> GeometryType:
+    raise TypeError(f"Unsupported type: {type(geometry)}")
+
+
+@to_fractions.register
+def _(geometry: Point) -> Point:
+    return Point(Fraction(geometry.x), Fraction(geometry.y))
+
+
+@to_fractions.register
+def _(geometry: Contour) -> Contour:
+    return Contour(list(map(to_fractions, geometry.vertices)))
+
+
+@to_fractions.register
+def _(geometry: Polygon) -> Polygon:
+    return Polygon(to_fractions(geometry.border),
+                   list(map(to_fractions, geometry.holes)))
+
+
+def union(*geometries: Shaped) -> Union[Shaped, Empty]:
+    if not geometries:
+        return EMPTY
+    return reduce(operator.or_, geometries)
+
+
+def midpoint(start: Point,
+             end: Point) -> Point:
+    """Returns coordinates of the middle of the input line"""
+    return Point((start.x + end.x) / 2, (start.y + end.y) / 2)
+
+
+def splitter_point(requirement: float,
+                   pivot: Point,
+                   low_area_point: Point,
+                   high_area_point: Point) -> Point:
+    """Alternative to bisection search since we always have triangles"""
+    if requirement <= 0:
+        raise ValueError("Can't have a zero or negative requirement")
+    p = pivot
+    l = low_area_point
+    h = high_area_point
+    contour = Contour([p, l, h])
+    triangle = Polygon(contour)
+    if triangle.area < requirement:
+        raise ValueError("Can't have a requirement greater than the area of "
+                         "the triangle")
+    r = requirement

```

Comment:
can we remove those cryptic abbreviations?

Reply:
I really don't like this function. I should think more later on how to refactor it. I'll change the abbreviations then.